Mathlib/Data/Nat/Interval.lean

```diff
@@ -396,4 +396,11 @@ theorem Nat.cauchy_induction_two_mul (seed : ℕ) (hs : P seed.succ)
   Nat.cauchy_induction_mul h 2 seed one_lt_two hs hm n
 #align nat.cauchy_induction_two_mul Nat.cauchy_induction_two_mul
 
+theorem Nat.pow_imp_self_of_one_lt {M} [Monoid M] (k : ℕ) (hk : 1 < k)
```

Comment:
suggestion : rename hsucc to hmul and hP to hpow

Reply:
also : given hsucc, hP seems automatic (if P x doesn't hold, then P (x ^ k) can't, by induction on k), or there's something that confuses me.

Reply:
done, thanks!